src/tif_model_iterator/tif_kernel_iterator.py

```diff
@@ -40,942 +38,309 @@ class tif_kernel_iterator_generator:
-
-        A fading kernel uses weights to give other pixel than the center pixel less weight in the prediction.
+        # Divide the satellite images into multiple parts and loop through the parts, using parts reduces the amount of RAM required to run this process.
+        for x_step in tqdm(range(begin_part, self.parts)):
+            self._create_and_write_part_output(
+                x_step=x_step, x_step_size=x_step_size, bottom=bottom, top=top
+            )
 
-        @param kernel: A kernel you which to  fade.
-        @return: A kernel that is faded now.
-        """
-        return kernel * self.fade_kernel
+        self._write_full_gdf_to_file()
+        self._clean_up_part_files()
 
-    def normalize_tile_kernel(self, kernel):
+    def _create_and_write_part_output(
+        self,
+        x_step: int,
+        x_step_size: int,
+        bottom: int,
+        top: int,
+    ):
         """
-        Normalize image kernels with sklearn's normalize.
-
-        @param kernel: a kernel to normalize
+        Creates the label predictions for a part of self.data and writes it to file
 
+        @param: x_step: Which step is currently being processed
+        @param: x_step_size: The step size for each step in the x direction
+        @param bottom: Lowest y coordinate
+        @param top: Highest y coordinate
         """
+        print("-------")
+        print("Part: " + str(x_step + 1) + " of " + str(self.parts))
+        left_boundary = x_step * x_step_size
+        right_boundary = (x_step + 1) * x_step_size
 
-        copy_kernel = np.zeros(shape=kernel.shape)
-        for x in range(0, kernel.shape[0]):
-            copy_kernel[x] = preprocessing.normalize(kernel[x])
+        subset_data = self.data[:, left_boundary:right_boundary, bottom:top]
 
-        return copy_kernel
+        subset_df = self._create_pixel_coordinate_dataframe(
+            data=subset_data, left_boundary=left_boundary
+        )
 
-    def normalize_min_max(self, kernel):
-        """
+        subset_df = self._filter_out_empty_pixels(subset_df)
 
-        Normalize tif file with min max scaler.
-        @param kernel: a kernel to normalize
+        # Check if a normalizer or a  scaler has to be used.
+        if self.normalize_scaler is not False:
+            print("Normalizing/Scaling data")
+            start = timer()
+            subset_df = self.normalize_scaler.transform(subset_df)
+            print(f"Normalizing/scaling finished in: {str(timer() - start)} second(s)")
 
-        """
+        subset_df = self._predict_labels(df=subset_df)
 
-        copy_kernel = np.zeros(shape=kernel.shape)
-        for x in range(0, kernel.shape[0]):
-            copy_kernel[x] = (
-                (kernel[x] - np.min(kernel[x]))
-                / (np.max(kernel[x]) - np.min(kernel[x]))
-                * 255
-            )
+        if self.aggregate_output:
+            subset_df = self._aggregate_pixel_labels(subset_df, new_pixel_size=2)
```

Comment:
this can be parameterized, the aggregate parameter size

Reply:
Yes, good addition!